### ISSUE TITLE
[bug + fix] - Wrong casing for `date` fields with MySQL

### ISSUE THREAD

**Issue by machour** [drizzle-team/drizzle-kit-mirror#46]
While introspecting my database, this schema got generated:

```ts
export const news = mysqlTable(
  "news",
  {
    id: int("id").autoincrement().primaryKey().notNull(),
    // you can use { mode: 'date' }, if you want to have Date as type for this column
    published_at: date("published_at", { mode: "string" }),
		// ... bunch of other fields
    createdAt: datetime("created_at", { mode: "string" }),
    updatedAt: datetime("updated_at", { mode: "string" }),
  },
  (table) => {
    return {
			// my indexes..
      idxNewsPublishedAt: index("idx_news_published_at").on(table.publishedAt),
    };
  }
);
```

Notice how the field name is still `published_at`, while it's referred to as `publishedAt` in the index.

This would fix it in the source code:

```diff
      if (lowered === "date") {
        let out = `// you can use { mode: 'date' }, if you want to have Date as type for this column
-       ${name}: date("${name}", { mode: 'string' })`;
+       ${name.camelCase()}: date("${name}", { mode: 'string' })`;
        defaultValue = defaultValue === "now()" ? ".defaultNow()" : defaultValue ? `.default(${mapColumnDefault(defaultValue)})` : "";
        out += defaultValue;
        return out;
      }
```

**Comment by AndriiSherman** [comment]
@machour 
Thanks a lot!
We are going to opensource drizzle-kit so you can make this contribution yourself!
But I can do it myself as well

**Comment by machour** [comment]
Hey @AndriiSherman 👋

Will happily contribue this fix (and many others) once the kit gets opensourced. Converting a bunch of PHP/MySQL projects to node, and I'm hitting a bunch of small issues ✌️

**Comment by AndriiSherman** [comment]
Fixed in `drizzle-kit@0.18.1`

I mentioned you as a contributor to a release
